GUI/test/javafxgui/controller/GUIAppControllerTest.java

```diff
@@ -20,62 +20,70 @@
 import static jttt.core.game.GameType.*;
 import static org.junit.Assert.assertEquals;
 
-public class TTTControllerTest {
+public class GUIAppControllerTest {
 
     private final int DEFAULT_BOARD_DIMENSION = 3;
     private final int GUI_WINDOW_HEIGHT = 700;
     private final int GUI_WINDOW_WIDTH = 600;
     private GUIViewSpy gameViewSpy;
-    private TTTController controller;
+    private GUIAppController controller;
     private GUIGameMaker gameMaker;
+    private GUIGameMakerFake fakeGameMaker;
 
     @Before
     public void setUp() {
         gameViewSpy = new GUIViewSpy(new Scene(new StackPane(), GUI_WINDOW_HEIGHT, GUI_WINDOW_WIDTH));
-        gameMaker = new GUIGameMaker(new GUIPlayerFactory());
-        controller = new TTTController(gameViewSpy, gameMaker);
+        gameMaker = new GUIGameMaker();
+        fakeGameMaker = new GUIGameMakerFake();
+        controller = new GUIAppController(gameMaker, gameViewSpy);
         new JFXPanel();
     }
 
     @Test
     public void startGameWithGameTypeAndDimension() {
-        Controller controller = new TTTController(gameViewSpy, new GUIGameMaker(new GUIPlayerFactory()));
-        controller.startGame(GUI_HVC, DEFAULT_BOARD_DIMENSION);
+        GUIAppController controller = new GUIAppController(new GUIGameMaker(), gameViewSpy);
+        controller.startGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
         assertEquals(Player.Type.GUI, controller.getCurrentPlayer().getType());
         assertEquals(false, gameViewSpy.hasResultBeenAnnounced());
     }
 
     @Test
     public void presentGameOptions() {
-        Controller controller = new TTTController(gameViewSpy, gameMaker);
-        controller.presentGameOptions();
+        GUIAppController controller = new GUIAppController(gameMaker, gameViewSpy);
+        controller.displayGameOptions();
         assertEquals(true, gameViewSpy.hasGameOptionsBeenPresented());
     }
 
     @Test
     public void displayGUI() {
-        controller.initializeGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
-        controller.displayGameLayout();
+        controller.startGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
         assertEquals(true, gameViewSpy.hasGameLayoutBeenRendered());
     }
 
     @Test
-    public void checkForResultUnsuccessful() {
+    public void noWinnerFound() {
         controller.initializeGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
         assertEquals(false, controller.foundWinOrDraw());
     }
 
     @Test
+    public void moveMadeByBoardClickStoredWithGUIHumanPlayer() {
+        controller.startGame(GUI_HVH, DEFAULT_BOARD_DIMENSION);
+        controller.registerPlayerMove("1");
+        GUIHumanPlayer humanPlayer = (GUIHumanPlayer)controller.getCurrentPlayer();
+        assertEquals(true, humanPlayer.hasNewMove());
```

Comment:
I judged that it didn't matter, as long as it had a move, but you're right, it could have another moved stored (though this should be disallowed because board should be disabled, but need to confirm))